CONTRIBUTING.md

```diff
@@ -0,0 +1,117 @@
+
+Changelogs are necessary so that the end user knows what changes were brought in each new release. This project maintains changelogs using the 
+[Keep a Changelog](https://keepachangelog.com/en/1.0.0/) guidelines.
+
+Version numbers are important in helping to differentiate releases of a product. This project follows [Semantic Versioning](https://semver.org/)
+to keep track of versions.
+
+Android uses a unique integer for each release called the Version Code. To ensure proper version code sequences and correlation with Semantic 
+Versioning, the following steps are used to arrive at the version code:
+
+1. The Major version shall be multipled by 1,000,000.
+
+2. The Minor version shall be multiplied by 10,000.
+
+3. The Patch version shall be multipled by 100.
+
+4. The above values shall be added together to get the version code a release.
+
+5. Pre-releases, alpha releases etc shoud be signified by appropriate values in the least two significant digits.
+
+6. The [changelog](CHANGELOG.md) should state the semantic version number as well as version code (if a Playstore rollout has been done) with each 
+   release.
+
+
+## Maintainers' Release Checklist
+
+There is no release timeline for this project. Once the maintainer(s) decide that sufficent changes have been made and any and all release blocking
+bugs have been quashed, a maintainer (identified as the releaser henceforth) will take charge of creating a release. The following checklist is to 
+be followed, and all work is to be done on `development` unless otherwise mentioned:
+
+
+1. Ensure that all relavant changes have been noted in the [changelog](CHANGELOG.md).
+
+2. Group the **Unreleased** changelogs under a new version number.
+   
+3. Update the `versionName` as well as `versionCode` in the app's [`build.gradle`](app/build.gradle) file.
+
+4. Ensure the above change does not result in the the debug and release build to fail.
+
+5. Create a commit with the aforementioned changes. Tag this commit with the version number using `git tag`.
+   Fast-forward `master` to the latest `development`.
+
+6.  Push both `master` and `development` directly to the repo. While doing so, the releaser should exercise 
+    utmost caution.  Use a PR if you are unsure and are doing this process for the first time. Make sure you push
+    the tag that you just created as well using `git push --tags <name-of-remote>`.
+   
+7. Draft any notes or posts that are to be put on any forum to notify users about the release.
+
+8. Compile a signed release apk using the apps signing key.
+
+9. Head over to the Google Play Store console and create a new release using the release apk.
+
+10. Add stub changelog notes for the next series of unreleased changes. Once again, create a commit. However, this time,
+    push only to `development`. 
+    
+**Note Regarding `master`**: `master` should always point to the latest release commit i.e a commit that updates the version number in the changelog and 
```

Comment:
I'm confused about why this was decided. It works, obviously, but if you have tags pointing to releases, you can continue development on master all the time instead of using a development branch.

I'd argue in favor of master branch just because that's the default and what people are used to. It would take additional overhead for contributors initially to get used to pushing to development.

Reply:
I wasn't there when the decision for using `master` and `development` was taken. So no comments there :P However, at this point, I don't think it's worthwhile getting rid of `development` and using `master` alone. I don't think there is significant additional overhead to new contributors either.